### PR TITLE
chore: Split library side pane for adding package control section

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
Purpose of this PR is to split the Libraries section to extend it in EE and introduced a tab layout that has both js libraries and packages
Changes:
1. Increased width of Libraries pane to 384px
2. Added a route `/package` which opens the same as library 
3. Extracted the JSLibraries section to be reused as a tab panel body in EE
4. Created `SidePaneWrapper` common component to be reused in EE
5. Split the title of the Libraries section under a hook to return empty string if package feature flag is enabled.
6. Minor linting refactors.

In CE there is only one visual change i.e the increase in width of the Libraries side pane from 250px to 384px

[Figma](https://www.figma.com/design/Z0QsSdGOydURn6WIMQ3rHM/Appsmith-Reusability?node-id=6816-319511&node-type=frame&t=x0DzN9HIxjefe3J1-0)

PR for https://github.com/appsmithorg/appsmith-ee/pull/5362
## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11434291701>
> Commit: ec76cc569ea90ba55e451f5ab18fa85244ac076f
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11434291701&attempt=3" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Mon, 21 Oct 2024 08:54:31 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Release Notes

- **New Features**
  - Introduced new functions for constructing URLs for application libraries and packages.
  - Added new route constants and matching functions for improved routing of application packages and libraries.
  - Implemented a custom hook for dynamic header titles in the libraries section.
  - Added a new `LibrarySidePane` component to enhance the editor interface.
  - Introduced a new `SidePaneWrapper` component for better layout management.

- **Bug Fixes**
  - Enhanced entity identification logic to recognize both libraries and packages.

- **Chores**
  - Updated routing logic for improved maintainability.
  - Added constants for configurable UI component widths.

These updates aim to enhance user experience and improve the overall functionality of the application.

### CODE REVIEW

File: app/client/src/IDE/Components/SidePaneWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+interface SidePaneContainerProps {
+  children?: ReactNode;
+  padded?: boolean;
+}
+
+const StyledContainer = styled(Flex)<Pick<SidePaneContainerProps, "padded">>`
+  padding: ${({ padded }) => padded && "var(--ads-v2-spaces-2)"};
+`;
+
+function SidePaneWrapper({ children, padded = false }: SidePaneContainerProps) {
```
Comment (sagar-qa007) [inline app/client/src/IDE/Components/SidePaneWrapper.tsx:15]: Do we need add/update any e2e or unit test case here?

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/src/IDE/Components/SidePaneWrapper.tsx (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/IDE/Components/SidePaneWrapper.tsx:15]: I don't believe that is required here. We do not check for css

File: app/client/src/ce/pages/Editor/IDE/Header/useLibraryHeaderTitle.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { createMessage, HEADER_TITLES } from "ee/constants/messages";
```
[the remark this answers is not in the document]
Comment (ashit-rath) [reply app/client/src/ce/pages/Editor/IDE/Header/useLibraryHeaderTitle.ts:1]: This is a known pattern being followed here

**Review by subrata71 — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**